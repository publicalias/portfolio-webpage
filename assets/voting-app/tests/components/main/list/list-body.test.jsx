"use strict";

//local imports

const ListBody = require("../../../../scripts/components/main/list/list-body");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("schemas/master");
const { newPoll } = require("schemas/voting-app");
const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//setup

beforeAll(reactTests.setup);

//list body

describe("list body", () => {

  const { testShallow } = testWrapper(ListBody);

  const testSnapshot = initTestSnapshot(testShallow);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (authenticated)", () => testSnapshot({ user: newUser() }));

  it("should match snapshot (polls)", () => testSnapshot({ polls: [newPoll()] }));

});
