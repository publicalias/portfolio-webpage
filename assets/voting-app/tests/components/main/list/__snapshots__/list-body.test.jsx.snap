// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list body should match snapshot (authenticated) 1`] = `
<div
  className="c-list-body"
>
  <div
    className="c-list-body__scroll-view js-scroll-view qa-scroll-view"
    onScroll={[Function]}
  >
    <table
      className="c-list-body__table js-scroll-content"
    >
      <thead>
        <tr>
          <td>
            <h5>
              Hide
            </h5>
          </td>
          <td>
            <h5>
              Flag
            </h5>
          </td>
          <td>
            <h5>
              Description
            </h5>
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            N/A
          </td>
          <td>
            N/A
          </td>
          <td>
            No Polls
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`list body should match snapshot (default) 1`] = `
<div
  className="c-list-body"
>
  <div
    className="c-list-body__scroll-view js-scroll-view qa-scroll-view"
    onScroll={[Function]}
  >
    <table
      className="c-list-body__table js-scroll-content"
    >
      <thead>
        <tr>
          <td>
            <h5>
              Hide
            </h5>
          </td>
          <td>
            <h5>
              Description
            </h5>
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            N/A
          </td>
          <td>
            No Polls
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`list body should match snapshot (polls) 1`] = `
<div
  className="c-list-body"
>
  <div
    className="c-list-body__scroll-view js-scroll-view qa-scroll-view"
    onScroll={[Function]}
  >
    <table
      className="c-list-body__table js-scroll-content"
    >
      <thead>
        <tr>
          <td>
            <h5>
              Hide
            </h5>
          </td>
          <td>
            <h5>
              Description
            </h5>
          </td>
        </tr>
      </thead>
      <tbody>
        <ListItem
          actions={
            Object {
              "formAddOption": [MockFunction],
              "formClearState": [MockFunction],
              "formRemoveOption": [MockFunction],
              "formSetAdd": [MockFunction],
              "formSetTitle": [MockFunction],
              "formToggleConfirm": [MockFunction],
              "formToggleSecret": [MockFunction],
              "listClearState": [MockFunction],
              "listSetSearch": [MockFunction],
              "metaAddErrors": [MockFunction],
              "metaCloseError": [MockFunction],
              "metaCreatePoll": [MockFunction],
              "metaDeletePoll": [MockFunction],
              "metaGetPolls": [MockFunction],
              "metaGetUser": [MockFunction],
              "metaNoOp": [MockFunction],
              "metaSetLoading": [MockFunction],
              "metaSetState": [MockFunction],
              "metaTimeoutError": [MockFunction],
              "pollAddOption": [MockFunction],
              "pollCastVote": [MockFunction],
              "pollRemoveOption": [MockFunction],
              "pollToggleFlag": [MockFunction],
              "pollToggleHide": [MockFunction],
              "pollToggleSecret": [MockFunction],
              "viewClearState": [MockFunction],
              "viewSetAdd": [MockFunction],
              "viewToggleConfirm": [MockFunction],
              "viewToggleSettings": [MockFunction],
            }
          }
          data={
            Object {
              "errors": Array [],
              "form": Object {
                "add": "",
                "confirm": false,
                "options": Array [],
                "secret": false,
                "title": "",
              },
              "list": Object {
                "search": "",
              },
              "loading": 0,
              "polls": Array [
                Object {
                  "author": "",
                  "date": 0,
                  "id": "",
                  "options": Array [],
                  "secret": false,
                  "title": "",
                  "users": Object {
                    "created": "",
                    "flagged": Array [],
                    "hidden": Array [],
                    "voted": 0,
                  },
                },
              ],
              "user": Object {},
              "view": Object {
                "add": "",
                "confirm": false,
                "settings": false,
              },
            }
          }
          history={
            Object {
              "push": [MockFunction],
            }
          }
          key="0-0"
          local={
            Object {
              "poll": Object {
                "author": "",
                "date": 0,
                "id": "",
                "options": Array [],
                "secret": false,
                "title": "",
                "users": Object {
                  "created": "",
                  "flagged": Array [],
                  "hidden": Array [],
                  "voted": 0,
                },
              },
            }
          }
          location={Object {}}
          match={Object {}}
          staticContext={Object {}}
        />
      </tbody>
    </table>
  </div>
</div>
`;
